diff -Nur gpsdrive-2.10pre7.orig/scripts/osm/perl_lib/CMakeLists.txt gpsdrive-2.10pre7/scripts/osm/perl_lib/CMakeLists.txt
--- gpsdrive-2.10pre7.orig/scripts/osm/perl_lib/CMakeLists.txt	2009-06-14 07:57:25.000000000 -0600
+++ gpsdrive-2.10pre7/scripts/osm/perl_lib/CMakeLists.txt	2009-06-17 11:13:56.000000000 -0600
@@ -16,7 +16,7 @@
         foreach(perl_module ${perl_modules})
           file(RELATIVE_PATH relative_module_path ${CMAKE_CURRENT_SOURCE_DIR} ${perl_module})
           get_filename_component(relative_module_path ${relative_module_path} PATH)
-          install(FILES ${perl_module} DESTINATION ${PERL_SITELIB}/${relative_module_path})
+          install(FILES ${perl_module} DESTINATION ${PERL_VENDORLIB}/${relative_module_path})
           #message("DEBUG: install ${perl_module} to ${PERL_SITELIB}/${relative_module_path}")
         endforeach(perl_module ${perl_modules})
       endif (perl_modules)
diff -Nur gpsdrive-2.10pre7.orig/scripts/osm/perl_lib/Geo/Gpsdrive/getstreet.pm gpsdrive-2.10pre7/scripts/osm/perl_lib/Geo/Gpsdrive/getstreet.pm
--- gpsdrive-2.10pre7.orig/scripts/osm/perl_lib/Geo/Gpsdrive/getstreet.pm	2009-06-14 07:57:25.000000000 -0600
+++ gpsdrive-2.10pre7/scripts/osm/perl_lib/Geo/Gpsdrive/getstreet.pm	2009-06-21 19:41:12.000000000 -0600
@@ -2,7 +2,7 @@
 #noch nicht alle variablen angepasst
 use LWP::UserAgent;
 use WWW::Mechanize;
-use Text::Query;
+#use Text::Query;
 use Getopt::Long; #maybe not needed any more
 use Pod::Usage;
 use DBI;
diff -Nur gpsdrive-2.10pre7.orig/scripts/osm/perl_lib/Geo/OSM/Upload.pm gpsdrive-2.10pre7/scripts/osm/perl_lib/Geo/OSM/Upload.pm
--- gpsdrive-2.10pre7.orig/scripts/osm/perl_lib/Geo/OSM/Upload.pm	2009-06-14 07:57:25.000000000 -0600
+++ gpsdrive-2.10pre7/scripts/osm/perl_lib/Geo/OSM/Upload.pm	2009-06-21 19:40:48.000000000 -0600
@@ -4,7 +4,7 @@
 use warnings;
 
 
-use WWW::Curl::easy;
+use WWW::Curl::Easy;
 
 sub new(){bless{}};
 
diff -Nur gpsdrive-2.10pre7.orig/src/gpsdrive_config.c gpsdrive-2.10pre7/src/gpsdrive_config.c
--- gpsdrive-2.10pre7.orig/src/gpsdrive_config.c	2009-06-14 07:57:25.000000000 -0600
+++ gpsdrive-2.10pre7/src/gpsdrive_config.c	2009-06-22 22:22:10.000000000 -0600
@@ -889,7 +889,7 @@ config_init ()
 	g_snprintf(local_config.mapnik_input_path, sizeof(local_config.mapnik_input_path),
 		"%s", "/usr/lib/mapnik/0.7/input/");
 	g_snprintf(local_config.mapnik_font_path, sizeof(local_config.mapnik_font_path),
-		"%s", "/usr/share/fonts/truetype/ttf-dejavu/");
+		"%s", "/usr/share/fonts/TTF/dejavu/");
 	
 	/* kismet default values */
 	g_strlcpy(local_config.kismet_servername, "127.0.0.1", sizeof(local_config.kismet_servername));
